File: 3. GODINA/2. SEMESTAR/Hardverski interfejsi/Domaci/Domaci 1/hi.jl
using ControlSystems, Plots, LinearAlgebra

function v_in(t)
    return abs.(sin.(314 .* t))
end

Rt = 10;
Rd = 2;
Rl = 1;
Rc = 1;
Rp = 100;
R = Rt + Rd + Rl;
L = 1e-3;
C = 60e-3;

a = (Rc*C*L)/(Rp);
b = Rc*C + (R*Rc*C)/(Rp) + L/Rp + L*C;
c = R*C + R/Rp + 1;

G  = tf([Rc*C, 1], [a, b, c]);
t = 0:0.01:100
#bodeplot(G)

w = L/Rp + L*C
println(1/w)
#Gsimple = tf([1], [L/Rp + L*C, 1])
#bodeplot(Gsimple)

y, t, x = lsim(G, (v_in(t))', t);
plot(t, y')

File: 3. GODINA/2. SEMESTAR/Hardverski interfejsi/Domaci/Domaci 1/hi.m
Rl = 1;
Rc = 1;
L = 100e-2;
C = 1e-5;
Rp = 100;

b1 = Rc*C;
b0 = 1;

a2 = L*(Rc*C/Rp + C);
a1 = (Rc*C + Rc*C/Rp + L/Rp);
a0 = Rl/Rp + 1;

G = tf([b1 b0], [a2 a1 a0]);
t = 0:0.01:1000;

bode(G)

%%
u = abs(sin(314*t));
y = lsim(G1, u, t);
hold on
plot(t, y)
plot(t, u)
